Example-QueryInfo-REXX/RXQUERY1.rexx

```diff
@@ -89,20 +102,162 @@ if RESULT <> 0 then
+/*******************************************************/
+GetLocalCPCInfo:
+
+methodSuccess = 1 /* assume true */
+emptyCPCResponse = '{"cpcs":[]}'
+localCPCfound = FALSE
+
+/* First list the cpcs, then retrieve local
+   CPC which will have
+           "location":"local"
+
+   GET /api/cpcs
+
+   v1: HWILIST(HWI_LIST_LOCAL_CPC)
+*/
+reqUri = '/api/cpcs'
+CPCInfoResponse = GetRequest(reqUri)
+
+emptyCPCArray = INDEX(CPCInfoResponse, emptyCPCResponse)
+if emptyCPCArray > 0 | CPCInfoResponse = '' Then
+  do
+    say 'fatalError ** failed to get CPC info **'
+    return 0
+  end
+
+/* Parse the response to obtain the uri
+   and target name associated with CPC,
+   which will be used to query storage info
+*/
+call JSON_parseJson CPCInfoResponse
+CPCArray = JSON_findValue(0, "cpcs", HWTJ_ARRAY_TYPE)
+if wasJSONValueFound(CPCArray) = FALSE then
+  do
+    say 'fatalError ** failed to retrieve CPCs **'
+    return 0
+  end
+
+/**************************************************************/
+ /* Determine the number of CPCs represented in the array     */
+ /**************************************************************/
+ CPCs = JSON_getArrayDim(CPCArray)
+ if CPCs <= 0 then
+    return fatalError( '** Unable to retrieve number of array entries **' )
+
+ /********************************************************************/
+ /* Traverse the CPCs array to populate CPCsList.                    */
+ /* We use the REXX (1-based) idiom  but adjust for the differing    */
+ /* (0-based) idiom of the toolkit.                                  */
+ /********************************************************************/
+ say 'Processing information for '||CPCs||' CPC(s)... searching for local'
+ do i = 1 to CPCs
+    nextEntryHandle = JSON_getArrayEntry(CPCArray,i-1)
+    CPClocation = JSON_findValue(nextEntryHandle,"location",HWTJ_STRING_TYPE)
+
+    if CPClocation = 'local' then
+      do
+        say 'found local CPC'
+        if localCPCfound <> FALSE then
+          do
+            say 'fatalError ** found two local CPCs **'
+            return 0
+          end
+        localCPCfound = TRUE
+
+        CPCuri=JSON_findValue(nextEntryHandle,"object-uri",HWTJ_STRING_TYPE)
+        if wasJSONValueFound(CPCuri) = FALSE then
+          do
+            say 'fatalError **failed to obtain CPC uri**'
+            return 0
+          end
+
+        CPCtargetName=JSON_findValue(nextEntryHandle,"target-name",HWTJ_STRING_TYPE)
+        if wasJSONValueFound(CPCtargetName) = FALSE then
+          do
+            say 'fatalError **failed to obtain CPC target name**'
+            return 0
+          end
+
+        CPCname=JSON_findValue(nextEntryHandle,"name",HWTJ_STRING_TYPE)
+        if wasJSONValueFound(CPCname) = FALSE then
+          do
+            say 'fatalError **failed to obtain CPC name**'
+            return 0
+          end
+      end
+ end /* endloop thru the JSON LPARs array */
+
+if localCPCfound = FALSE then
+  do
+    methodSuccess = FALSE
```

Comment:
Is FALSE an actual boolean value in REXX, and if so why don't we set methodSuccess to TRUE at the beginning of this method instead of 1?